server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -520,23 +530,41 @@ public Clusters(
          * @param total total number of clusters in the search
          * @param successful number of successful clusters in the search
          * @param skipped number of skipped clusters (skipped can only happen for remote clusters with skip_unavailable=true)
+         * @param running number of running clusters in the search
+         * @param partial number of partial response from clusters in the search
+         * @param failed number of failed clusters
          */
-        public Clusters(int total, int successful, int skipped) {
-            // TODO: change assert to total == 1 or total = 0 - this should probably only be used for local searches now
+        public Clusters(int total, int successful, int skipped, int running, int partial, int failed) {
```

Comment:
Will running, partial and failed will always be zero? I think yes because this constructor is only used for local-only searches (other than maybe some tests) and in that case we don't track running/partial/failed states, right? 

If they will always be zero, then I don't think we need to add the new counters to the constructor - just default them to zero.

Reply:
Yes, you are right, in this case will always be zero. I modified the constructor accordingly.